CMakeLists.txt

```diff
@@ -880,9 +880,17 @@ include(CheckCXXCompilerFlag)
 add_subdirectory(bindings)
 
 if(webtorrent)
-target_link_libraries(torrent-rasterbar PRIVATE LibDataChannel::LibDataChannelStatic)
-target_include_directories(torrent-rasterbar PRIVATE deps/json/include)
-target_compile_definitions(torrent-rasterbar PRIVATE BOOST_JSON_HEADER_ONLY)
+	target_link_libraries(torrent-rasterbar PRIVATE LibDataChannel::LibDataChannelStatic)
+
+	# Boost.JSON was added to Boost in version 1.75
+	find_package(Boost OPTIONAL_COMPONENTS json)
+	if(Boost_JSON_FOUND)
+		target_link_libraries(torrent-rasterbar PUBLIC ${Boost_JSON_LIBRARY})
+	else()
+		# Fallback to the submodule
+		target_compile_definitions(torrent-rasterbar PRIVATE BOOST_JSON_HEADER_ONLY)
```

Comment:
when we're using it as header-only, we need to include `src.hpp` too, don't we?

Reply:
I have a feeling that it might be easier to use boost.json as header-only until libtorrent can require boost >= 1.75, and unconditionally rely on it from boost. But I could be wrong.

Reply:
Oh, good point, the way to use the library header-only changed in the doc from defining `BOOST_JSON_HEADER_ONLY` to including `boost/json/src.hpp`. I'll fix that.

Reply:
Using the library all the time would simplify the compilation setup a bit but I'm afraid I'll conflict with the one present in Boost.

Reply:
right. I posted this PR against your PR. I may have misunderstood the header-only use then.

https://github.com/paullouisageneau/libtorrent/pull/1

(feel free to squash my commit in btw)

Reply:
I think the reason this worked in CI was because there is no cmake build where webtorrent is enabled. feel free to enable it. I think maybe appveyor is the only remaining cmake coverage on CI right now

Reply:
Actually, webtorrent is already enabled for the msvc build on appveyor.

Reply:
this is the command line I see executed on appveyor:

```
cmake -DBOOST_LIBRARYDIR=%BOOST_ROOT%\stage\lib -DCMAKE_CXX_STANDARD=17 -Dbuild_tests=ON -Dbuild_examples=ON -Dbuild_tools=ON -Dpython-bindings=%python% -Dboost-python-module-name="python" -Dskip-python-runtime-test=true -DPython_ADDITIONAL_VERSIONS="2.7" -G "Visual Studio 16 2019" -A x64 .. && cmake --build . --config Release --parallel %NUMBER_OF_PROCESSORS% -- -verbosity:minimal
```

And in the cmake output, I see:

```
-- The following features have been disabled:
 * static_runtime, build libtorrent with static runtime
 * python-bindings, build python bindings
 * python-egg-info, generate python egg info
 * python-install-system-dir, Install python bindings to the system installation directory rather than the CMake installation prefix
 * gnutls, build using GnuTLS instead of OpenSSL
 * webtorrent, build with WebTorrent support
```

Reply:
Indeed, the option is passed to b2 but not to cmake.